Client/Frontend/Browser/BrowserToolbar.swift

```diff
@@ -36,102 +30,72 @@ class BrowserToolbar: UIView, UITextFieldDelegate, BrowserLocationViewDelegate {
 
     override init(frame: CGRect) {
         super.init(frame: frame)
-        viewDidInit()
+        initViews()
     }
 
     required init(coder aDecoder: NSCoder) {
         super.init(coder: aDecoder)
-        viewDidInit()
     }
 
-    private func viewDidInit() {
-        self.backgroundColor = UIColor(white: 0.80, alpha: 1.0)
+    func addButtons(buttons: UIButton...) {
+        for button in buttons {
+            button.setTitleColor(UIColor.blackColor(), forState: UIControlState.Normal)
+            button.setTitleColor(UIColor.grayColor(), forState: UIControlState.Disabled)
+            button.imageView?.contentMode = UIViewContentMode.ScaleAspectFit
+            addSubview(button)
+        }
+
+        var prev: UIView? = nil
+        for view in self.subviews {
```

Comment:
The way this works right now, this view basically just lays out its subviews (in layoutSubviews now instead of in addButtons) to all take equal space inside its frame. It doesn't care what the subviews are. Its a little like a UIToolbar I guess, except without the abstraction between the UIBarButtonItem and the UIViews (and without support for spacers).

Do you still want to handle non-button subviews differently?